contracts/staking/BaseShareField.sol

```diff
@@ -56,41 +56,50 @@ contract BaseShareField is DAOContract{
         shareToken = _shareToken;
     }
 
-    // Update reward variables of the given pool to be up-to-date.
+    /**
+    @dev Update reward variables of the given pool to be up-to-date. 
+    * Calculates passed blocks and adding to the reward pool
+     */
     function _update() internal virtual {
         if (totalProductivity == 0) {
             lastRewardBlock = block.number;
             return;
         }
         FundManager fm = FundManager(nameService.getAddress("FUND_MANAGER"));
-        (uint rewardsPerBlock, uint blockStart, uint blockEnd, bool isBlackListed) = fm.getStakingReward(address(this));
+        (uint rewardsPerBlock, uint blockStart, uint blockEnd, ) = fm.rewardsForStakingContract(address(this));
+        if(block.number >= blockStart && lastRewardBlock < blockStart) lastRewardBlock = blockStart;
         if(block.number >= blockStart && blockEnd>= block.number){
             uint256 multiplier = block.number - lastRewardBlock;
             uint256 reward = multiplier * (rewardsPerBlock * 1e16); // turn it to 18 decimals
```

Comment:
the comment should be
rewardsPerBlock is in G$ which is only 2 decimals, we turn it into 18 decimals